Minio/DataModel/Select/SelectResponseStream.cs

```diff
@@ -0,0 +1,256 @@
+                {
+                    Array.Reverse(preludeCRCBytes); 
+                }
+                numBytesRead += n;
+                var inputArray = new byte[prelude.Length + 4 ];
+                System.Buffer.BlockCopy(prelude,0,inputArray,0, prelude.Length);
+
+                // write real data to inputArray
+                Crc32Algorithm.ComputeAndWriteToEnd(inputArray); // last 4 bytes contains CRC
+                // transferring data or writing reading, and checking as final operation
+                if (!Crc32Algorithm.IsValidWithCrcAtEnd(inputArray))
+                {
+                    throw new ArgumentException("invalid prelude CRC");
+                }
+
+                if (!Enumerable.SequenceEqual(inputArray.Skip(prelude.Length).Take(4), preludeCRCBytes))
+                {
+                    throw new ArgumentException("Prelude CRC Mismatch");
+                }
+                 var bytes = prelude.Take(4).ToArray();
+                 if (BitConverter.IsLittleEndian)
+                 {
+                    Array.Reverse(bytes); 
+                 }
+                int totalLength = BitConverter.ToInt32(bytes, 0);
+                bytes = prelude.Skip(4).Take(4).ToArray();
+                if (BitConverter.IsLittleEndian)
+                {
+                    Array.Reverse(bytes); 
+                }
+
+                int headerLength = BitConverter.ToInt32(bytes, 0);
+                int payloadLength = totalLength - headerLength - 16;
+
+                var headers = new byte[headerLength];
+                var payload  = new byte[payloadLength];
+                int num = ReadFromStream(headers);
+                if (num != headerLength)
+                {
+                    throw new IOException("insufficient data");
+
+                }
+                num = ReadFromStream(payload);
+                if (num != payloadLength)
+                {
+                    throw new IOException("insufficient data");
+                }
+
+                numBytesRead += num;
+                num = ReadFromStream(messageCRC);
+                var messageCRCBytes = messageCRC.ToArray();
+                if (BitConverter.IsLittleEndian)
+                {
+                    Array.Reverse(messageCRCBytes); 
+                }
+                // now verify message CRC
+                inputArray = new byte[totalLength];
+                System.Buffer.BlockCopy(prelude,0,inputArray,0, prelude.Length);
+                System.Buffer.BlockCopy(preludeCRC,0,inputArray,prelude.Length, preludeCRC.Length);
+                System.Buffer.BlockCopy(headers,0,inputArray,prelude.Length+ preludeCRC.Length, headerLength);
+                System.Buffer.BlockCopy(payload,0,inputArray,prelude.Length+ preludeCRC.Length+headerLength, payloadLength);
+
+                // write real data to inputArray
+                Crc32Algorithm.ComputeAndWriteToEnd(inputArray); // last 4 bytes contains CRC
```

Comment:
So, this overwrites the last four bytes of the payload. `inputArray = new byte[totalLength+4];` may fix this.

Reply:
@Praveenrajmani , totalLength includes the entire 16 byte overhead including 4 for the messageCRC. The messageCRC is not copied into the inputArray, leaving 4 bytes for the CRC32 algorithm to store the computed value. Seems correct to me.